src/lib/loadInstructions.js

```diff
@@ -10,47 +21,116 @@ function readYamlFile(p) {
   return yaml.load(txt);
 }
 
+/*
+Parse a match bit string (e.g., "0100000----------101-----0010011")
+into an array of single characters, each representing one bit.
+
+Returns null if the string isn’t 16 or 32 bits long.
+*/
 function parseMatchBits(matchStr) {
-  if (!matchStr || matchStr.length !== 32) return null;
+  if (!matchStr) return null;
+  if (matchStr.length !== 32 && matchStr.length !== 16) return null;
   return matchStr.split("");
 }
 
+/*
+Parse a bitfield location string into numeric segments.
+Example:
+  "31-25|11-7" → [ {from:31,to:25}, {from:11,to:7} ]
+
+Each segment defines one contiguous region of bits, and fields can have
+multiple disjoint segments (like immediates in S- or B-type instructions).
+*/
+function parseLocationSegments(loc) {
+  if (loc === undefined || loc === null) return [];
```

Comment:
As above, why would `loc` ever be `undefined` or `null`? Do we really need to handle this case? Allowing the function to silently continue on malformed input seems like it will make this function harder to debug, because problems will be less obvious.

Reply:
Looks like this is still relevant. Can we just delete this?